stacks-node/src/burnchains/rpc/bitcoin_rpc_client/mod.rs

```diff
@@ -0,0 +1,510 @@
+    ///
+    /// Wallet is stored in the `-walletdir` specified in the Bitcoin Core configuration (or the default data directory if not set).
+    ///
+    /// # Arguments
+    /// * `wallet_name` - Name of the wallet to create.
+    /// * `disable_private_keys` - If `Some(true)`, the wallet will not be able to hold private keys.
+    ///   If `None`, this defaults to `false`, allowing private key import/use.
+    ///
+    /// # Returns
+    /// Returns `Ok(())` if the wallet is created successfully.
+    ///
+    /// # Availability
+    /// - **Since**: Bitcoin Core **v0.17.0**.
+    ///
+    /// # Notes
+    /// This method supports a subset of available RPC arguments to match current usage.
+    /// Additional parameters can be added in the future as needed.
+    pub fn create_wallet(
+        &self,
+        wallet_name: &str,
+        disable_private_keys: Option<bool>,
+    ) -> BitcoinRpcClientResult<()> {
+        let disable_private_keys = disable_private_keys.unwrap_or(false);
+
+        self.global_ep.send::<Value>(
+            &self.client_id,
+            "createwallet",
+            vec![wallet_name.into(), disable_private_keys.into()],
+        )?;
+        Ok(())
+    }
+
+    /// Returns a list of currently loaded wallets by the Bitcoin Core node.
+    ///
+    /// # Returns
+    /// A vector of wallet names as strings.
+    ///
+    /// # Availability
+    /// Available since Bitcoin Core **v0.15.0**.
+    pub fn list_wallets(&self) -> BitcoinRpcClientResult<Vec<String>> {
+        Ok(self
+            .global_ep
+            .send(&self.client_id, "listwallets", vec![])?)
+    }
+
+    /// Retrieve a list of unspent transaction outputs (UTXOs) that meet the specified criteria.
+    ///
+    /// # Arguments
+    /// * `min_confirmations` - Minimum number of confirmations required for a UTXO to be included (Default: 0).
+    /// * `max_confirmations` - Maximum number of confirmations allowed (Default: 9.999.999).
+    /// * `addresses` - Optional list of addresses to filter UTXOs by (Default: no filtering).
+    /// * `include_unsafe` - Whether to include UTXOs from unconfirmed unsafe transactions (Default: `true`).
+    /// * `minimum_amount` - Minimum amount (in BTC. As String to preserve full precision) a UTXO must have to be included (Default: "0").
+    /// * `maximum_count` - Maximum number of UTXOs to return. Use `None` for effectively unlimited (Default: 9.999.999).
+    ///
+    /// # Returns
+    /// A Vec<[`ListUnspentResponse`]> containing the matching UTXOs.
+    ///
+    /// # Notes
+    /// This method supports a subset of available RPC arguments to match current usage.
+    /// Additional parameters can be added in the future as needed.
+    pub fn list_unspent(
+        &self,
+        min_confirmations: Option<u64>,
+        max_confirmations: Option<u64>,
+        addresses: Option<&[&str]>,
+        include_unsafe: Option<bool>,
+        minimum_amount: Option<&str>,
```

Comment:
Please use a u64 to represent an numerical quantity, not a `&str`.  The method can covert this to a `&str` internally.